lib/services/base-path-mapping-service.js

```diff
@@ -0,0 +1,166 @@
+"use strict";
+
+const { ApiGwVersions } = require("../constants");
+const ServerlessService = require("./serverless-service");
+
+class BasePathMappingService extends ServerlessService {
+	constructor(serverless, options, logger) {
+		super(serverless, options, logger);
+	}
+
+	async createBasePathMappingAsync() {
+		const mappingExists = await this._mappingExistsAsync(this.config.basePath, this.config.customDomainName);
+		if (!mappingExists) {
+			this.logger.log(`Creating base path mapping ${this.config.basePath} for domain name ${this.config.customDomainName}...`);
+			return this._createMappingAsync();
+		}
+
+		this.logger.log(`Base path mapping ${this.config.basePath} for domain name ${this.config.customDomainName} already exist. It will NOT be recreated.`);
+		return true;
+	}
+
+	async removeBasePathMappingAsync() {
+		const mappingExists = await this._mappingExistsAsync(this.config.basePath, this.config.customDomainName);
+		if (!mappingExists) {
+			this.logger.log(`Base path mapping ${this.config.basePath} for domain name ${this.config.customDomainName} does NOT exist.`);
+			return true;
+		}
+
+		this.logger.log(`Removing base path mapping ${this.config.basePath} from domain name ${this.config.customDomainName}...`);
+		return this._removeMappingAsync(this.config.customDomainName, this.config.basePath);
+	}
+
+	checkRequiredConfigProperties(config) {
+		const errors = super.checkRequiredConfigProperties(config);
+
+		if (config && !config.basePath) {
+			errors.push("Missing 'basePath' in the plugin configuration.");
+		}
+
+		return errors;
+	}
+
+	async _mappingExistsAsync(basePath, domainName) {
+		const mappings = await this.provider.request("APIGateway", "getBasePathMappings", { domainName });
+		if (!mappings || !mappings.items || !mappings.items.length) {
+			return false;
```

Comment:
If `v1`  `apiGateway` does not have mappings wi will not check for `v2` mappings.

As a side node - this could be: 
```

	async _mappingExistsAsync(basePath, domainName) {
		const mappingV1 = this._getMappingInfoV1Async(domainName, basePath);
		const mappingV2 = this._getMappingInfoV2Async(domainName, basePath);

		return !!mappingV1 || !!mappingV2;
	}

	async _getMappingInfoV1Async(domainName, basePath) {
		return this._getMappingInfoAsync("APIGateway", "getBasePathMappings", { domainName }, (i) => i.basePath === basePath)
	}

	async _getMappingInfoV2Async(domainName, basePath) {
		return this._getMappingInfoAsync("ApiGatewayV2", "getApiMappings", { DomainName: domainName }, (i) => i.ApiMappingKey === basePath);
	}

	async _getMappingInfoAsync(apiGatewayVersion, method, options, filter) {
		const mappings = await this.provider.request(apiGatewayVersion, method, options);
		if (!mappings || !mappings.Items || !mappings.Items.length) {
			return null;
		}

		return mappings.Items.find(filter);
	}
```

Reply:
Done.